src/validations/html/rules/MismatchedTag.js

```diff
@@ -0,0 +1,139 @@
+import {List, Map, Record, Set} from 'immutable';
```

Comment:
This module uses containers keyed by locations. (AFAIK) the native JavaScript `Map` and `Set` cannot support custom equality for user-defined types. Their keys may be JavaScript objects but the native containers compare them by identity and (AFAIK) do not support custom equality.

So this uses `Map` and `Set` keyed by the `Record` type `Location`. This could've converted the `{row, col}` to string keys but I figured using a more direct `Record` type would be cleaner and more strongly-typed.